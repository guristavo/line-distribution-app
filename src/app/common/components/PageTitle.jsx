import React from 'react';
import PropTypes from 'prop-types';

const PageTitle = ({ title, isBeta, isAdmin }) => (
  <h1 className="page-title">
    {title}
    {isAdmin && (
      <span className="page-title__flag page-title__flag--admin">Admin</span>
    )}
    {isBeta && (
      <span className="page-title__flag page-title__flag--beta">Beta</span>
    )}
  </h1>
);

PageTitle.defaultProps = {
  isAdmin: false,
  isBeta: false,
};

PageTitle.propTypes = {
  title: PropTypes.string.isRequired,
  isAdmin: PropTypes.bool,
  isBeta: PropTypes.bool,
};

export default PageTitle;
