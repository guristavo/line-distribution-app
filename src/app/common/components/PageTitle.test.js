import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import PageTitle from './PageTitle';

describe('common/PageTitle', () => {
  const TRUE = true;

  afterEach(cleanup);

  it('renders', () => {
    const { asFragment } = render(<PageTitle title="Page Title" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders the admin banner', () => {
    const { getByText } = render(
      <PageTitle title="Page Title" isAdmin={TRUE} />
    );
    expect(getByText('Admin')).toBeInTheDocument();
  });

  it('renders the beta banner', () => {
    const { getByText } = render(
      <PageTitle title="Page Title" isBeta={TRUE} />
    );
    expect(getByText('Beta')).toBeInTheDocument();
  });
});
