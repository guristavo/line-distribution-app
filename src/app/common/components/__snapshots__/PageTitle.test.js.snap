// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common/PageTitle renders 1`] = `
<DocumentFragment>
  <h1
    class="page-title"
  >
    Page Title
  </h1>
</DocumentFragment>
`;
